### PR TITLE
Support build/publish one module via TargetModule property

### PR DESCRIPTION
**Pull request by dingmeng-xue**
## Description

TargetModule can be set by user to just build and publish one module and its dependency Az.Accounts. For instance
```
dotnet msbuild build.proj /t:"Build;Publish" /p:"Configuration=Release;NuGetKey=12345;TargetModule=Compute"
```

It cannot support the array of modules because existing code in publish has limitation and assumes Az always needs to be published. I check if Az is required according to number of module is built. If it is more than 2, it means user doesn't use TargetModule property. For long term, we need to find a smart way to check if Az needs to be published according to the content in artifact folder only.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks/SetGenerator.cs [hunk 1]
```diff
@@ -152,5 +120,37 @@ public static HashSet<string> Generate(HashSet<string> filesChangedSet, Dictiona
...
+        /// <param name="moduleNames"></param>
+        /// <param name="mappingsDictionary"></param>
+        /// <returns></returns>
+        public static HashSet<string> Generate(string[] moduleNames, Dictionary<string, string[]> mappingsDictionary)
+        {
+            var outputSet = new HashSet<string>();
+
+            foreach (var module in moduleNames)
+            {
+                foreach (var pair in mappingsDictionary)
```
Comment (erich-wang) [inline tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks/SetGenerator.cs:135]: Suggest to use mappingDictionary.ContainsKey($"src/{module}/")

Comment (dingmeng-xue) [reply tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks/SetGenerator.cs]: Updated.

**Pull request merged**